packages/camera/camera/ios/Classes/CameraPermissionUtils.m

```diff
@@ -5,35 +5,83 @@
 @import AVFoundation;
 #import "CameraPermissionUtils.h"
 
-void FLTRequestCameraPermissionWithCompletionHandler(
-    FLTCameraPermissionRequestCompletionHandler handler) {
-  switch ([AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeVideo]) {
+void FLTRequestPermission(BOOL forAudio, FLTCameraPermissionRequestCompletionHandler handler) {
+  AVMediaType mediaType;
+  if (forAudio) {
+    mediaType = AVMediaTypeAudio;
+  } else {
+    mediaType = AVMediaTypeVideo;
+  }
+
+  switch ([AVCaptureDevice authorizationStatusForMediaType:mediaType]) {
```

Comment:
I know `authorizationStatusForMediaType:mediaType` was already used before, but should this be `+requestAccessForMediaType:completionHandler:` instead to actually request it?

Reply:
Oh we should always check the status first, and only request the permission if it's in `.notDetermined` status.